File: cpp/raindrops/raindrops.cpp
#include "raindrops.h"

namespace raindrops {
std::string convert(int n) {
    std::string res = "";
    if (n % 3 == 0) res += "Pling";
    if (n % 5 == 0) res += "Plang";
    if (n % 7 == 0) res += "Plong";
    if (res == "") res = std::to_string(n);
    return res;
}
}  // namespace raindrops

File: zig/raindrops/raindrops.zig
const io = @import("std").io;

pub fn convert(buffer: []u8, n: u32) []const u8 {
    var buff = io.fixedBufferStream(buffer);
    const writer = buff.writer();
    if (n % 3 == 0) writer.writeAll("Pling") catch unreachable;
    if (n % 5 == 0) writer.writeAll("Plang") catch unreachable;
    if (n % 7 == 0) writer.writeAll("Plong") catch unreachable;
    if (buff.pos == 0) writer.print("{d}", .{n}) catch unreachable;
    return buffer[0..buff.pos];
}
